services/core-api/app/api/utils/models_mixins.py

```diff
@@ -82,26 +86,47 @@ def deep_update_from_dict(self, data_dict, depth=0):
         Return: None
         Side-Effect: Self attributes have been overwritten by values in data_dict, matched on key, recursivly.
         """
-
         current_app.logger.debug(depth * '-' + f'updating{self}')
         model = inspect(self.__class__)
         editable_columns = [
             c for c in model.columns if c.name not in [pk.name for pk in model.primary_key]
         ]
+        if not _edit_key:
+            if not self._edit_key:
+                current_app.logger.warn(
+                    'Model._edit_key not set, no related models will be updated')
+            _edit_key = self._edit_key
         assert isinstance(data_dict, dict)
         for k, v in data_dict.items():
             current_app.logger.debug(depth * '>' + f'{type(v)}-{k}')
             if isinstance(v, dict):
+                rel = getattr(self.__class__, k)
+                rel_class = rel.property.entity.class_
+                if _edit_key not in rel_class._edit_groups:
+                    current_app.logger.debug(
+                        f'COMBOBREAKER!!! {_edit_key} not in {rel_class} edit groups {rel_class._edit_groups}'
+                    )
+                    continue
                 current_app.logger.debug(depth * ' ' + f'recursivly updating {k}')
-                getattr(self, k).deep_update_from_dict(v, depth=(depth + 1))
+                getattr(self, k).deep_update_from_dict(v, depth=(depth + 1), _edit_key=_edit_key)
 
             if isinstance(v, list):
+                rel = getattr(self.__class__, k) #SA.relationship definition
                 obj_list = getattr(self, k)
+                obj_list_class = rel.property.entity.class_
+
+                if _edit_key not in obj_list_class._edit_groups:
+                    #not part of same edit group, skip
+                    current_app.logger.debug(
+                        f'COMBOBREAKER!!! {_edit_key} not in {obj_list_class} edit groups {obj_list_class._edit_groups}'
```

Comment:
Do we want to keep these debug statements in?

Reply:
Yes, but could use be tweaked, debug is the highest detail, we should use err,warn,info,debug logging levels better.